README.md

```diff
@@ -50,7 +52,8 @@ Here’s a complete `package.json` example, for a hypothetical module named `@mo
   "exports": {
     "./": "./src/util/",
     "./timezones/": "./data/timezones/",
-    "./timezones/utc": "./data/timezones/utc/index.mjs"
+    "./timezones/utc": "./data/timezones/utc/index.mjs",
+    "./core-polyfill": ["std:core-module", "./core-polyfill"]
```

Comment:
we should probably also have an example that uses an optional dep - like chokidar or something - since that’s currently a common use case for file-watching utilities.

Reply:
I think that's its own discussion, worth a separate issue. Right now we only do static validation and won't try to fetch all candidates, in series. Given that a package manager could know at install-time that an optional dependency wasn't installed, it feels something better left to the package manager.

Reply:
Are you suggesting that at install time, the package manager would rewrite an observable part of package.json? Because i think that’d be terrible and I’d advocate strongly against any package manager ever doing that.

I’d expect it to function in the same way as std:whatever - if it’s resolveable, grab it, else move to the next.

Reply:
also there’s no fetching; in node all deps are present on disk.

Reply:
Disk access is fetching, lower latency but still fetching. And since its IO, it can be exposed to all the weirdness that IO brings. Access errors, temporary failures, race conditions in changing contents, ...

Reply:
in terms of validation sure, but std:whatever would require the same disk fetching - so either this example should be removed, or an actual real-world example of common and identical behavior with an optional dep should be added.

Reply:
I wouldn't expect that `std:whatever` hits any kind of IO. That should be a statically known list of supported `std:` modules. If you prefer, we could use a more "realistic" straw person name of a built-in module and/or protocol but that feels like we'd invite distractions.

Reply:
I don't think it makes sense to privilege builtin modules over userland ones; if we're not going to validate the presence of userland ones at this stage, we shouldn't be validating the presence of anything.

Reply:
To nitpick: We're not validating the presence of an implementation but the fact that we recognize this kind of URL. I'd also assume that we ignore `data:` URLs with unsupported encodings or network protocols that we don't support like `ftp:`.

Reply:
in that case, makes perfect sense to me for an early validation error.

Reply:
Is there any further action to be taken on this one?

Reply:
i still think it’d be helpful to show an actual real world example with a common optional dep like chokidar, alongside this one.